src/orion/core/worker/trial.py

```diff
@@ -167,13 +166,20 @@ def __init__(self, **kwargs):
                 for item in value:
                     attr.append(self.Param(**item))
             else:
-                setattr(self, attrname, value)
+                if attrname == '_id':
+                    if self.params is not None and self.experiment is not None:
+                        self._id = self.id
+                else:
+                    setattr(self, attrname, value)
 
     def to_dict(self):
         """Needed to be able to convert `Trial` to `dict` form."""
         trial_dictionary = dict()
 
         for attrname in self.__slots__:
+            if attrname == '_id':
+                continue
+
```

Comment:
Stuff like this are code smells. See total review.

Reply:
I agree, it hurt a little writing this